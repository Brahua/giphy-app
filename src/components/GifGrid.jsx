import React from 'react'
import GifGridItem  from './GifGridItem';
import { useFetchGif } from '../hooks/useFetchGif';
import PropTypes from 'prop-types';

const GifGrid = ({category}) => {
  
  const {data:gifs, loading} = useFetchGif(category);


  return (
    <div className="row mx-3 mx-sm-0">
      <h3 className="title-section w-100 animate__animated animate__fadeIn">{category}</h3>
      { loading && 
        <div className="spinner-border text-info my-3" role="status">
          <span className="sr-only">Loading...</span>
        </div>
      }
      <div className="card-columns">  
        { gifs.map(gif => <GifGridItem key={gif.id} {...gif}></GifGridItem> )}
      </div> 
    </div>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}


export default GifGrid

