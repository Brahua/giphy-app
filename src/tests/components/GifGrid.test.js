import React from 'react';
import '@testing-library/jest-dom'
import {shallow} from 'enzyme';
import GifGrid from './../../components/GifGrid'
import { useFetchGif } from './../../hooks/useFetchGif';
jest.mock('./../../hooks/useFetchGif');

describe('Tests in <GifGrid>', () => {
  const component = <GifGrid category={'DBZ'}></GifGrid>;


  test('should create the component correctly', () => {
    useFetchGif.mockReturnValue({
      data: [],
      loading: true
    })
    const wrapper = shallow(component);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show gifs information correctly', () => {
    const gifs = [{
      id: 'ABC',
      url: 'https://localhost/cualquier/cosa.jpg',
      title: 'Cualquier cosa'
    },
    {
        id: '123',
        url: 'https://localhost/cualquier/cosa.jpg',
        title: 'Cualquier cosa'
    }];

    useFetchGif.mockReturnValue({
      data: gifs,
      loading: false
    })

    const wrapper = shallow(component);
    expect(wrapper.find('span').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
})
