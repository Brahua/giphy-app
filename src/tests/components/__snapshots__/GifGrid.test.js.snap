// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGrid> should create the component correctly 1`] = `
<div
  className="row mx-3 mx-sm-0"
>
  <h3
    className="title-section w-100 animate__animated animate__fadeIn"
  >
    DBZ
  </h3>
  <div
    className="spinner-border text-info my-3"
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
  <div
    className="card-columns"
  />
</div>
`;
